src/server/gc_worker/compaction_filter.rs

```diff
@@ -690,21 +691,24 @@ pub mod test_utils {
         pub start: Option<&'a [u8]>,
         pub end: Option<&'a [u8]>,
         pub target_level: Option<usize>,
-        pub gc_scheduler: FutureScheduler<GcTask<RocksEngine>>,
+        pub gc_scheduler: Scheduler<GcTask<RocksEngine>>,
         pub gc_receiver: UnboundedReceiver<Option<GcTask<RocksEngine>>>,
         pub(super) callbacks_on_drop: Vec<Arc<dyn Fn(&WriteCompactionFilter) + Send + Sync>>,
     }
 
     impl<'a> TestGCRunner<'a> {
         pub fn new(safe_point: u64) -> Self {
-            let (tx, rx) = unbounded();
+            let (_tx, rx) = unbounded();
+            let worker = WorkerBuilder::new("test-gc-runner").create();
+            let gc_scheduler = worker.lazy_build("gc-runner").scheduler();
+
             TestGCRunner {
                 safe_point,
                 ratio_threshold: None,
                 start: None,
                 end: None,
                 target_level: None,
-                gc_scheduler: FutureScheduler::new("test-gc-sched", tx),
+                gc_scheduler,
                 gc_receiver: rx,
```

Comment:
Should be `worker.receiver`?

Reply:
Do you mean implement `pub fn receiver(&self) -> Option<UnboundedReceiver<Msg<T>>>` for `LazyWorker`?